src/atoms/previewAtoms.ts

```diff
@@ -3,4 +3,40 @@ import { atom } from "jotai";
 
 export const selectedComponentsPreviewAtom = atom<ComponentSelection[]>([]);
 
+export const visualEditingSelectedComponentAtom =
+  atom<ComponentSelection | null>(null);
+
+export const currentComponentCoordinatesAtom = atom<{
+  top: number;
+  left: number;
+  width: number;
+  height: number;
+} | null>(null);
+
 export const previewIframeRefAtom = atom<HTMLIFrameElement | null>(null);
+
+export interface PendingStyleChange {
```

Comment:
i'm wondering if this type can be reused with the IPC type (this seems like a subset of the type used in the visual_editing_handlers